Sources/TuistCache/Utilities/CacheXCFrameworkBuilder.swift

```diff
@@ -182,6 +238,29 @@ public final class CacheXCFrameworkBuilder: CacheArtifactBuilding {
     fileprivate func frameworkPath(fromArchivePath archivePath: AbsolutePath, productName: String) -> AbsolutePath {
         archivePath.appending(RelativePath("Products/Library/Frameworks/\(productName).framework"))
     }
+
+    fileprivate func isMacCatalystSupported(projectPath: AbsolutePath, projectName: String) -> Bool {
```

Comment:
The target/project is defined by tuist, do we really need to use xcodebuild to have this info?

Reply:
Thank you for providing the context. 🙏 
I also don't think that this approach is the best option. I'll think about it again. 🤔

Reply:
I explored a tuist today to improve this approach. To improve this approach, I believe that we need to check if MacCatalyst is supported using the parameters received as shown in the log in the image below. To do this, it is expected that the relevant value for `XCFramework` creation by tuist should be passed as a parameter. However,  in my opinion, to modify this, I think would need to make extensive changes to the codebase. What do you think? Is there a better way?

*here the log image*
<img width="2240" alt="Pasted Graphic 18" src="https://user-images.githubusercontent.com/45344633/225857571-d31aba66-c81c-4d2c-83a9-0f33620c2d7b.png">

Reply:
The `builld` method is called in the `CacheController` that has access to the `Graph` instance, wouldn't be enough to pass that down to the `CacheXCFrameworkBuilder` to have access to the target information?

Reply:
I wasn't aware that we could retrieve the value through the `Graph` instance of the `CacheController`. Thank you for follow-up! I've updated the logic to retrieve the value through the `Graph` instance and check it accordingly.